nonce/proto/nonce.proto

```diff
@@ -0,0 +1,19 @@
+syntax = "proto2";
+
+package nonce;
+option go_package = "proto";
+
+import "core/proto/core.proto";
+
+service NonceService {
+	rpc Nonce(core.Empty) returns (NonceMessage) {}
+	rpc Valid(NonceMessage) returns (ValidMessage) {}
```

Comment:
Bikeshedding: As long as we're tweaking let's call this method "Redeem" rather than "Valid" to indicate that it changes state.